src/app/ipv/middleware.js

```diff
@@ -64,8 +64,16 @@ async function fetchUserDetails(req) {
   return generateUserDetails(userDetailsResponse, req.i18n);
 }
 
-async function handleJourneyResponse(req, res, action) {
-  const backendResponse = (await journeyApi(action, req)).data;
+async function handleJourneyResponse(req, res, action, currentPage = "") {
+
+  if (currentPage) {
+    action += `?currentPage=${currentPage}`
+  }
```

Comment:
I'm hoping that Mike's changes can incorporate this, or if he merges in first then I will update to use his logic